src/plugins/csp_configuration_provider/README.md

```diff
@@ -0,0 +1,64 @@
+# CspConfigurationProvider
+
+A OpenSearch Dashboards plugin
+
+This plugin introduces the support of self service dynamic configuration of Content Security Policy (CSP) rules without requiring a server restart. It registers a pre-response handler to `HttpServiceSetup` which can get CSP rules from a new index `.opensearch_dashboards_config` and then rewrite to CSP header. OSD users could update the content of the index with new CSP rules and the change will take effect immediately.
```

Comment:
Do we have to store the config into an index? If taking security plugin into consideration, some users may not be able to get the correct config because of lack of permission and fallback to default value. Then different users may have different Csp config.